src/lsm/scan_tree.zig

```diff
@@ -271,6 +288,73 @@ pub fn ScanTreeType(
                 },
                 .needs_data => return error.ReadAgain,
                 .buffering => unreachable,
+                .aborted => return null,
+            }
+        }
+
+        /// Probing a scan modifies the key_min/key_max range, either moving the scan if the next
+        /// key is already buffered, or causing it to read again from the new key range.
+        pub fn probe(self: *ScanTree, key: Key) void {
```

Comment:
Also maybe rename `key` to `probe_key`. "key" is a very generic term, so being more specific can hint at its purpose.